cli/cmd/compose/build.go

```diff
@@ -65,6 +67,11 @@ func buildCommand(p *projectOptions) *cobra.Command {
 	cmd.Flags().MarkHidden("compress") //nolint:errcheck
 	cmd.Flags().Bool("force-rm", true, "Always remove intermediate containers. DEPRECATED")
 	cmd.Flags().MarkHidden("force-rm") //nolint:errcheck
+	cmd.Flags().BoolVar(&opts.noCache, "no-cache", false, "Do not use cache when building the image")
+	cmd.Flags().Bool("no-rm", false, "Do not remove intermediate containers after a successful build. DEPRECATED")
+	cmd.Flags().MarkHidden("no-rm") //nolint:errcheck
+	cmd.Flags().StringVarP(&opts.memory, "memory", "m", "", "Set memory limit for the build container. DEPRECATED")
```

Comment:
AFAIK this hasn't been deprecated (why would it?)
docker-compose 1.28.5 says `-m, --memory MEM        Set memory limit for the build container.` - no deprecation notice here

Reply:
I don't think they're deprecated, but some of these are not (yet) supported on BuildKit; there's a list of options that are not available when using buildkit here; https://github.com/moby/moby/issues/40379

Reply:
right, so we should better show a warning explaining this option is not (yet) implemented and will be ignored.

Reply:
On the docker cli, we conditionally hide some of these if buildkit is used as builder (look for the `no-buildkit` annotation for these flags; https://github.com/docker/cli/blob/e3d93058fdbb7ecd24aa68ca710908e075a0f2ed/cli/command/image/build.go#L123-L124

Reply:
> right, so we should better show a warning explaining this option is not (yet) implemented and will be ignored.

Yes, I had a PR in the cli, that would produce an error, but possibly a "warning" would be better; I was waiting for @tonistiigi to answer he agreed on that (instead of error); https://github.com/docker/cli/pull/2736

Reply:
while @aiordache is on PTO, I added a warning if the `--memory` flag is set, and it's just ignored. Removed `DEPRECATED` from the description (but it's hidden anyway)